serde/src/main/java/com/amazon/ionhiveserde/objectinspectors/map/AbstractFieldNameObjectInspector.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionhiveserde.objectinspectors.map;
+
+import com.amazon.ionhiveserde.objectinspectors.AbstractIonPrimitiveJavaObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+
+/**
+ * Base class for primitive object inspectors that need to handle overflow detection.
```

Comment:
So there is support for char and varchar via the FieldNameToStringObjectInspector - since string, varchars, and chars are all backed by java strings, we are just routing them in the ObjectInspectorFactory.

The exact location it's handled:
IonObjectInspectorFactory.java - Line 357
```
                    case VARCHAR:
                    case CHAR:
                    case STRING:
                        objectInspector = FIELD_NAME_TO_STRING_OBJECT_INSPECTOR;
                        break;
```

Reply:
Thanks, I tried to find logics in FieldNameToStringObjectInspector to validate each of them (e.g Varchar types are created with a length specifier between 1 and 65535) but seems that we are handling String, Char and Varchar equivalently in Ion-hive-serde layer.

For example, type `Varchar`:
> Varchar types are created with a length specifier (between 1 and 65535), which defines the maximum number of characters allowed in the character string.

or type `Char`:
> Char types are similar to Varchar but they are fixed-length meaning that values shorter than the specified length value are padded with spaces but trailing spaces are not important during comparisons.


The reason I bring it up here is just in case this is something we need to take care because the doucment mentioned `Like string, trailing whitespace is significant in varchar and will affect comparison results.`

My initial idea was creating some boolean like overflow check for numeric value, but it might be unnecessary since Hive might just simply truncate the exceeding part (or do we need to let users know that their string will be cut?).